### PR TITLE
Update to Java 17

### PR DESCRIPTION
**Pull request by findepi**
This also includes airbase update, as a requirement.

### CODE REVIEW

**Review by martint — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -34,6 +34,10 @@
 
     <properties>
         <air.test.jvmsize>2g</air.test.jvmsize>
+        <project.build.targetJdk>17</project.build.targetJdk>
+        <air.java.version>17.0.4</air.java.version>
+        <air.modernizer.java-version>8</air.modernizer.java-version>
+        <air.check.skip-pmd>true</air.check.skip-pmd>
```
Comment (kokosing) [inline pom.xml:40]: What is this?

File: pom.xml [hunk 2]
```diff
@@ -34,6 +34,10 @@
 
     <properties>
         <air.test.jvmsize>2g</air.test.jvmsize>
+        <project.build.targetJdk>17</project.build.targetJdk>
+        <air.java.version>17.0.4</air.java.version>
```
Comment (kokosing) [inline pom.xml:38]: why not 17.0.5?

**Review by kokosing — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (findepi) [reply pom.xml:40]: Good catch. PMD check failed to run in some weird way (don't remember details). I don't think I should care about it, so i disabled it.
Do you think i should keep PMD here?

File: pom.xml (hunk 2 above)
Comment (findepi) [reply pom.xml:38]: This doesn't matter, does it?

replaced with just `17`

**Comment by findepi** [comment]
(just rebased)

**Comment by wendigo** [comment]
Superseded by https://github.com/trinodb/tpch/pull/37

**Pull request closed without merging**